components/service/glean/src/test/java/mozilla/components/service/glean/GleanTest.kt

```diff
@@ -241,6 +242,67 @@ class GleanTest {
         }
     }
 
+    @Test
+    fun `test sending of metrics ping`() {
+        val server = MockWebServer()
+        server.enqueue(MockResponse().setBody("OK"))
+
+        EventsStorageEngine.clearAllStores()
+        val metricToSend = StringMetricType(
+                disabled = false,
+                category = "telemetry",
+                lifetime = Lifetime.Ping,
+                name = "metrics_ping",
+                sendInPings = listOf("default")
+        )
+
+        val metricsPingScheduler = MetricsPingScheduler(getContextWithMockedInfo())
+
+        // Store a ping timestamp from 25 hours ago to compare to so that the ping will get sent
+        metricsPingScheduler.updateSentTimestamp(
+                System.currentTimeMillis() - TimeUnit.HOURS.toMillis(25))
+
+        resetGlean(getContextWithMockedInfo(), Glean.configuration.copy(
+                serverEndpoint = "http://" + server.hostName + ":" + server.port,
+                logPings = true
+        ))
+
+        // Fake calling the lifecycle observer.
+        val lifecycleRegistry = LifecycleRegistry(mock(LifecycleOwner::class.java))
+        val gleanLifecycleObserver = GleanLifecycleObserver()
+        lifecycleRegistry.addObserver(gleanLifecycleObserver)
+
+        try {
+            // Simulate the first foreground event after the application starts.
+            lifecycleRegistry.handleLifecycleEvent(Lifecycle.Event.ON_RESUME)
+
+            // Write a value to the metric to check later
+            metricToSend.set("Test metrics ping")
+
+            // Simulate going to background so that the metrics ping schedule will be checked
+            lifecycleRegistry.handleLifecycleEvent(Lifecycle.Event.ON_STOP)
+
+            val requests: MutableMap<String, String> = mutableMapOf()
```

Comment:
Mh, I think you should also be receiving a `baseline` ping here along with the `metrics` ping. In order to avoid intermittent failures in the future, can you please check the ping type before moving on?

Reply:
It is receiving both a `metrics` ping and a `baseline` ping, but since it's creating a `mutableMapOf()` the pings are maped by the doc_type which is why `requests["metrics"]` would be null if this were a baseline ping.  I'm okay with digging in and checking the ping type but I followed the function that checks the background pings and it doesn't verify the ping types either (aside from looking for things that would only be in those ping types).

Reply:
Ah, I see. Thanks for clarifying, makes sense.!